### DISCUSSION TITLE
Visible percentage in pie chart

### DISCUSSION THREAD

**Discussion by buutqn** [chartjs/chartjs-plugin-datalabels#205] [Q&A] [answered]
I have the following options: 
```
datalabels: {
    formatter: (value, ctx) => {
        let sum = 0;
        let dataArr = ctx.chart.data.datasets[0].data;
        dataArr.map(data => {
            sum += data;
        });
        let percentage = (value * 100 / sum);
        if (percentage > 2.5) {
            return percentage.toFixed(2) + "%";
        } else {
            return "";
        }
    },
    color: "#fff",
}
```

I have a chart with several values, and many of them with low percentage values
![pie-chart-example](https://user-images.githubusercontent.com/15644387/99348308-10ebaf80-2878-11eb-83bd-68ae93e7ef26.png)

What i want is, i believe the formatter runs when a slice is shown or hidden, and, how can i calculate the percentage of the slice based on the shown slices? For example, when i hide (clicking on the legend) some slices, the small ones get biggers, and i'd like to show the label on them.
![pie-chart-hidden-example](https://user-images.githubusercontent.com/15644387/99348495-7d66ae80-2878-11eb-9ec9-a31b6dc9dd18.png)
Is there a way to get the current percentage of the size of the slice?

**Comment by simonbrunel** [comment, marked as answer]
@buutqn I moved your ticket as a discussion since it's neither a bug or feature request.

This plugin doesn't provide any way to know the effective percentage, it's internally computed by the core library. Also, the plugin doesn't expose any geometry info about the element. You could use the `chart.getDatasetMeta(datasetIndex)` API to retrieve meta data and use `meta.data[index].hidden` to know about data visibility but you would still need to compute the total yourself. 

Here is [an example](https://jsfiddle.net/simonbrunel/jzbae56g/2/).
